rust/ql/test/query-tests/unusedentities/unreachable.rs

```diff
@@ -0,0 +1,140 @@
+
+//fn cond() -> bool;
+//fn get_a_number() -> i32;
+
+// --- unreachable code --
+
+fn do_something() {
+}
+
+fn unreachable_if() {
+	if false {
+		do_something(); // BAD: unreachable code [NOT DETECTED]
```

Comment:
Should be fixed by https://github.com/github/codeql/pull/17602.

Reply:
Indeed this is now fixed, as are a couple of other cases when I merged in latest main.  :)